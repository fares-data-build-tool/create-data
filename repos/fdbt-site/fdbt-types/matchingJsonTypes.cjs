"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.ExpiryUnit = exports.CarnetExpiryUnit = void 0;
var CarnetExpiryUnit;
(function (CarnetExpiryUnit) {
    CarnetExpiryUnit["HOUR"] = "hour";
    CarnetExpiryUnit["DAY"] = "day";
    CarnetExpiryUnit["WEEK"] = "week";
    CarnetExpiryUnit["MONTH"] = "month";
    CarnetExpiryUnit["YEAR"] = "year";
    CarnetExpiryUnit["NO_EXPIRY"] = "no expiry";
})(CarnetExpiryUnit = exports.CarnetExpiryUnit || (exports.CarnetExpiryUnit = {}));
var ExpiryUnit;
(function (ExpiryUnit) {
    ExpiryUnit["HOUR"] = "hour";
    ExpiryUnit["DAY"] = "day";
    ExpiryUnit["WEEK"] = "week";
    ExpiryUnit["MONTH"] = "month";
    ExpiryUnit["YEAR"] = "year";
})(ExpiryUnit = exports.ExpiryUnit || (exports.ExpiryUnit = {}));
