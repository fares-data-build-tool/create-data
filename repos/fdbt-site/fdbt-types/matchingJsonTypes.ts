export interface PointToPointCarnetProductDetails extends BaseProduct {
    productName: string;
    carnetDetails: CarnetDetails;
}

export type FlatFareGeoZoneTicket = Omit<PeriodGeoZoneTicket, 'products' | 'type'> & {
    type: 'flatFare';
    products: FlatFareProductDetails[];
};

export interface PeriodGeoZoneTicket extends BasePeriodTicket {
    zoneName: string;
    stops: Stop[];
}

export interface BasePeriodTicket extends BaseTicket<'period' | 'multiOperator'> {
    operatorName: string;
    products: ProductDetails[];
    fareDayEnd?: string;
}

export interface SelectedService {
    lineName: string;
    lineId: string;
    serviceCode: string;
    startDate: string;
    serviceDescription: string;
}

export interface ServiceWithNocCode extends SelectedService {
    nocCode: string;
    selected: boolean;
    destination?: string;
    origin?: string;
}

export interface SelectedServiceByNocCode {
    [key: string]: ServiceWithNocCode[];
}

export interface AdditionalService extends SelectedService {
    serviceId: number;
}

export interface PeriodMultipleServicesTicket extends BasePeriodTicket {
    selectedServices: SelectedService[];
    termTime: boolean;
}

export interface FlatFareProductDetails extends BaseProduct {
    productName: string;
    productPrice: string;
    carnetDetails?: CarnetDetails;
}

export interface FlatFareMultipleServices extends BaseTicket<'flatFare'> {
    products: FlatFareProductDetails[];
    termTime: boolean;
    selectedServices: SelectedService[];
    operatorName: string;
}

export type FlatFareTicket = FlatFareGeoZoneTicket | FlatFareMultipleServices;

export interface SalesOfferPackage {
    id: number;
    name: string;
    description?: string;
    purchaseLocations: string[];
    paymentMethods: string[];
    ticketFormats: string[];
    price?: string;
}

export type FromDb<T> = T & { id: number };

export type TicketType = 'flatFare' | 'period' | 'multiOperator' | 'schoolService' | 'single' | 'return';

export type Ticket =
    | PointToPointTicket
    | GeoZoneTicket
    | PeriodMultipleServicesTicket
    | FlatFareTicket
    | SchemeOperatorGeoZoneTicket
    | SchemeOperatorFlatFareTicket
    | SchemeOperatorMultiServiceTicket
    | MultiOperatorGeoZoneTicket
    | PointToPointPeriodTicket
    | PeriodHybridTicket;

export type TicketWithIds =
    | WithIds<SingleTicket>
    | WithIds<ReturnTicket>
    | WithIds<GeoZoneTicket>
    | WithIds<PeriodMultipleServicesTicket>
    | WithIds<FlatFareTicket>
    | WithIds<SchemeOperatorGeoZoneTicket>
    | WithIds<SchemeOperatorFlatFareTicket>
    | WithIds<SchemeOperatorMultiServiceTicket>
    | WithIds<MultiOperatorMultipleServicesTicket>
    | WithIds<MultiOperatorGeoZoneTicket>
    | WithIds<PointToPointPeriodTicket>
    | WithIds<PeriodHybridTicket>;

export type GeoZoneTicket = PeriodGeoZoneTicket | MultiOperatorGeoZoneTicket;

export interface PeriodHybridTicket extends PeriodGeoZoneTicket, PeriodMultipleServicesTicket {}

export interface SchemeOperatorGeoZoneTicket extends BaseSchemeOperatorTicket {
    zoneName: string;
    stops: Stop[];
    products: ProductDetails[];
    additionalNocs: string[];
}

export interface SchemeOperatorFlatFareTicket extends BaseSchemeOperatorTicket {
    type: 'flatFare';
    products: FlatFareProductDetails[];
    additionalOperators: {
        nocCode: string;
        selectedServices: SelectedService[];
    }[];
}

export interface SchemeOperatorMultiServiceTicket extends BaseSchemeOperatorTicket {
    type: 'period';
    products: ProductDetails[];
    additionalOperators: {
        nocCode: string;
        selectedServices: SelectedService[];
    }[];
}

export interface MultiOperatorGeoZoneTicket extends PeriodGeoZoneTicket {
    additionalNocs: string[];
    operatorGroupId: number;
}

export interface MultiOperatorMultipleServicesTicket extends PeriodMultipleServicesTicket {
    additionalOperators: {
        nocCode: string;
        selectedServices: SelectedService[];
    }[];
    operatorGroupId: number;
}

export interface BaseTicket<T extends TicketType = TicketType> {
    nocCode: string;
    type: T;
    passengerType: string;
    ageRange?: string;
    ageRangeMin?: string;
    ageRangeMax?: string;
    proof?: string;
    proofDocuments?: string[];
    email: string;
    uuid: string;
    timeRestriction: FullTimeRestriction[];
    ticketPeriod: TicketPeriod;
    groupDefinition?: GroupDefinition;
    carnet?: boolean;
}

export type WithIds<T extends { products: BaseProduct[] }> = WithBaseIds<
    Omit<T, 'products'> & {
        products: (Omit<T['products'][0], 'salesOfferPackages'> & {
            salesOfferPackages: { id: number; price: string | undefined }[];
        })[];
    }
>;

export type WithBaseIds<T> = Omit<
    T,
    | 'passengerType'
    | 'ageRange'
    | 'ageRangeMin'
    | 'ageRangeMax'
    | 'proof'
    | 'proofDocuments'
    | 'timeRestriction'
    | 'fareDayEnd'
> & { passengerType: { id: number }; timeRestriction?: { id: number } };

export interface GroupDefinition {
    maxPeople: string;
    companions: CompanionInfo[];
}

export interface CompanionInfo {
    id: number;
    name?: string;
    passengerType: string;
    minNumber?: string;
    maxNumber: string;
    ageRangeMin?: string;
    ageRangeMax?: string;
    proofDocuments?: string[];
}

export interface DbTimeBand {
    startTime: string;
    endTime: string | { fareDayEnd: boolean };
}

export interface UnassignedStop {
    atcoCode: string;
}

export interface UnassignedStops {
    singleUnassignedStops?: UnassignedStop[];
    outboundUnassignedStops?: UnassignedStop[];
    inboundUnassignedStops?: UnassignedStop[];
}

export interface BaseSchemeOperatorTicket extends Omit<BaseTicket, 'nocCode'> {
    schemeOperatorName: string;
    schemeOperatorRegionCode: string;
}

export type SchemeOperatorTicket =
    | SchemeOperatorMultiServiceTicket
    | SchemeOperatorGeoZoneTicket
    | SchemeOperatorFlatFareTicket;

export interface BasePointToPointTicket extends BaseTicket {
    operatorName: string;
    lineName: string;
    lineId: string;
    serviceDescription: string;
    products: (BaseProduct | PointToPointCarnetProductDetails)[];
    unassignedStops: UnassignedStops;
}

export type PointToPointTicket = SingleTicket | ReturnTicket;

export interface SingleTicket extends BasePointToPointTicket {
    type: 'single';
    fareZones: FareZone[];
    termTime: boolean;
    journeyDirection: string;
}

export interface MatchingJsonMetaData {
    productId: string;
    serviceId?: string;
    matchingJsonLink: string;
}

export interface ReturnTicket extends BasePointToPointTicket {
    type: 'return';
    inboundFareZones: FareZone[];
    outboundFareZones: FareZone[];
    returnPeriodValidity?: ReturnPeriodValidity;
    additionalServices?: AdditionalService[];
}

export interface Product {
    productName: string;
    productPrice?: string;
    productDuration?: string;
    productValidity?: string;
    productDurationUnits?: string;
    productEndTime?: string;
    carnetDetails?: CarnetDetails;
}

export interface ProductDetails extends BaseProduct {
    productName: string;
    productPrice: string;
    productDuration: string;
    productValidity: string;
    carnetDetails?: CarnetDetails;
}

export interface PointToPointPeriodTicket extends Omit<ReturnTicket, 'products' | 'type'> {
    type: 'period';
    products: (PointToPointPeriodProduct & Product & BaseProduct)[];
}

export type TimeRestrictionDay =
    | 'monday'
    | 'tuesday'
    | 'wednesday'
    | 'thursday'
    | 'friday'
    | 'saturday'
    | 'sunday'
    | 'bankHoliday';

export interface FullTimeRestriction {
    day: TimeRestrictionDay;
    timeBands: TimeBand[];
}

export interface TicketPeriod {
    startDate: string;
    endDate?: string;
}

export interface BaseProduct {
    salesOfferPackages: SalesOfferPackage[];
}

export interface CarnetProductInfo {
    productName: string;
    carnetDetails: CarnetDetails;
}

export interface PointToPointPeriodProduct {
    productName: string;
    productDuration: string;
    productDurationUnits: ExpiryUnit;
}

export interface PeriodExpiry {
    productValidity: string;
    productEndTime?: string;
}

export interface FareZone {
    name: string;
    stops: Stop[];
    prices: FareZonePrices[];
}

export interface FareZonePrices {
    price: string;
    fareZones: string[];
}

export interface CarnetDetails {
    quantity: string;
    expiryTime: string;
    expiryUnit: CarnetExpiryUnit;
}

export interface TimeBand {
    startTime: string;
    endTime: string;
}

export interface ReturnPeriodValidity {
    amount: string;
    typeOfDuration: string;
}

export enum CarnetExpiryUnit {
    HOUR = 'hour',
    DAY = 'day',
    WEEK = 'week',
    MONTH = 'month',
    YEAR = 'year',
    NO_EXPIRY = 'no expiry',
}

export enum ExpiryUnit {
    HOUR = 'hour',
    DAY = 'day',
    WEEK = 'week',
    MONTH = 'month',
    YEAR = 'year',
    TERM = 'term',
}

export interface Stop {
    stopName: string;
    naptanCode: string;
    atcoCode: string;
    localityCode: string;
    localityName: string;
    parentLocalityName: string;
    qualifierName?: string;
    indicator?: string;
    street?: string;
}

export interface OperatorDetails {
    operatorName: string;
    contactNumber: string;
    email: string;
    url: string;
    street: string;
    town: string;
    county: string;
    postcode: string;
}
