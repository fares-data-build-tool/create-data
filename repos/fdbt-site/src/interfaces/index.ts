import { NextApiRequest, NextPageContext } from 'next';
import { DocumentContext } from 'next/document';
import { IncomingMessage, ServerResponse } from 'http';
import { ReactElement } from 'react';

// Session Attributes and Cookies

export interface Session {
    session: Express.Session;
}

export type NextApiRequestWithSession = NextApiRequest & Session;

export type NextPageContextWithSession = NextPageContext & {
    req: Session;
};

export type DocumentContextWithSession = DocumentContext & {
    req: Session;
};

export type IncomingMessageWithSession = IncomingMessage & Session;

export interface CookiePolicy {
    essential: boolean;
    usage: boolean;
}

export interface ProductInfo {
    productName: string;
    productPrice: string;
}

export interface ProductInfoWithErrors extends ProductInfo {
    errors: ErrorInfo[];
}

export interface DurationValidInfo {
    amount: string;
    duration: string;
    errors: ErrorInfo[];
}

export interface InputMethodInfo {
    inputMethod: string;
}

export interface Journey extends JourneyWithErrors {
    directionJourneyPattern?: string;
    inboundJourney?: string;
    outboundJourney?: string;
}

export interface JourneyWithErrors {
    errors?: ErrorInfo[];
}

export interface TicketRepresentationAttribute {
    name: string;
}

export interface TicketRepresentationAttributeWithErrors {
    errors: ErrorInfo[];
}

export interface Operator {
    name: string;
    nocCode: string;
}

export interface OperatorGroup {
    name: string;
    operators: Operator[];
}

export interface MultiOperatorInfo {
    nocCode: string;
    services: string[];
}

export interface TermTimeAttribute {
    termTime: boolean;
}

export type WithErrors<T> = {
    errors: ErrorInfo[];
} & T;

export interface UserDataUploadsProps {
    csvUploadApiRoute: string;
    csvUploadHintText: string;
    csvUploadTitle: string;
    guidanceDocDisplayName: string;
    guidanceDocAttachmentUrl: string;
    guidanceDocSize: string;
    csvTemplateDisplayName: string;
    csvTemplateAttachmentUrl: string;
    csvTemplateSize: string;
    errors: ErrorInfo[];
    detailSummary?: string;
    detailBody?: ReactElement;
    showPriceOption?: boolean;
    poundsOrPence?: string | null;
    csrfToken: string;
}

export interface GroupDefinitionWithErrors extends GroupDefinition {
    errors: ErrorInfo[];
}

export interface NumberOfStagesAttributeWithError {
    errors: ErrorInfo[];
}

export interface FareStagesAttribute {
    fareStages: string;
}

export interface FareStagesAttributeWithErrors {
    errors: ErrorInfo[];
}

export interface CsvUploadAttributeWithErrors {
    errors: ErrorInfo[];
    poundsOrPence?: string;
}

export interface GroupTicketAttribute {
    maxGroupSize: string;
}

export interface GroupTicketAttributeWithErrors extends GroupTicketAttribute {
    errors: ErrorInfo[];
}

export interface NumberOfProductsAttribute {
    numberOfProductsInput: string;
}

export interface NumberOfProductsAttributeWithErrors {
    errors: ErrorInfo[];
}

export interface MultipleProductAttribute {
    products: MultiProduct[];
}

export interface MultipleProductAttributeWithErrors extends MultipleProductAttribute {
    errors: ErrorInfo[];
}

export interface SchoolFareTypeAttribute {
    schoolFareType: string;
}

export interface MultipleOperatorsAttribute {
    selectedOperators: Operator[];
}

export interface MultipleOperatorsAttributeWithErrors extends MultipleOperatorsAttribute {
    errors: ErrorInfo[];
}

export interface ServiceListAttribute {
    selectedServices: string[];
}

export interface ServiceListAttributeWithErrors {
    errors: ErrorInfo[];
}

// Miscellaneous

export interface PassengerAttributes {
    passengerTypeDisplay: string;
    passengerTypeValue: string;
}

export interface BaseReactElement {
    id: string;
    name: string;
    label: string;
    defaultValue?: string;
    defaultChecked?: boolean;
    options?: string[];
}

export interface InputCheck {
    error: string;
    input: string;
    id: string;
}

export interface ErrorInfo {
    errorMessage: string;
    id: string;
    userInput?: string;
}

export interface Feedback {
    question: string;
    answer: string;
}

export interface PriceEntryError {
    input: string;
    locator: string;
}

export interface ResponseWithLocals extends ServerResponse {
    locals: {
        nonce: string;
        csrfToken: string;
    };
}

export interface PremadeTimeRestriction {
    name: string;
    contents: FullTimeRestriction[];
}

// AWS and Reference Data (e.g. NOC, TNDS, NaPTAN datasets)

export interface Stop {
    stopName: string;
    naptanCode: string;
    atcoCode: string;
    localityCode: string;
    localityName: string;
    parentLocalityName: string;
    qualifierName?: string;
    indicator?: string;
    street?: string;
}

export interface StopPoint {
    stopPointRef: string;
    commonName: string;
}

export interface S3NetexFile {
    name: string;
    noc: string | undefined;
    reference: string;
    fareType: string;
    productNames?: string;
    passengerType: string;
    serviceNames?: string;
    lineName?: string;
    zoneName?: string;
    sopNames: string;
    date: string;
    signedUrl: string;
    fileSize: number;
}

/* eslint-disable camelcase */
export interface CognitoIdToken {
    sub: string;
    aud: string;
    email_verified: boolean;
    event_id: string;
    'custom:noc': string;
    token_use: string;
    auth_time: number;
    iss: string;
    'cognito:username': string;
    exp: number;
    iat: number;
    email: string;
    'custom:contactable': string;
    'custom:schemeOperator': string;
    'custom:schemeRegionCode': string;
}

// Ticket Types

export type Ticket =
    | PointToPointTicket
    | GeoZoneTicket
    | PeriodMultipleServicesTicket
    | FlatFareTicket
    | SchemeOperatorTicket;

export type PointToPointTicket = SingleTicket | ReturnTicket;

export interface BaseTicket {
    nocCode: string;
    type: string;
    passengerType: string;
    ageRange?: string;
    ageRangeMin?: string;
    ageRangeMax?: string;
    proof?: string;
    proofDocuments?: string[];
    email: string;
    uuid: string;
    timeRestriction: FullTimeRestriction[];
    ticketPeriod: TicketPeriod;
}

export interface BasePointToPointTicket extends BaseTicket {
    operatorShortName: string;
    lineName: string;
    serviceDescription: string;
    products: BaseProduct[];
}

export interface SingleTicket extends BasePointToPointTicket {
    fareZones: FareZone[];
    termTime: boolean;
}

export interface ReturnTicket extends BasePointToPointTicket {
    inboundFareZones: FareZone[];
    outboundFareZones: FareZone[];
    returnPeriodValidity?: ReturnPeriodValidity;
}

export type PeriodTicket = PeriodGeoZoneTicket | PeriodMultipleServicesTicket;

export interface BasePeriodTicket extends BaseTicket {
    operatorName: string;
    products: ProductDetails[];
}

export interface PeriodGeoZoneTicket extends BasePeriodTicket {
    zoneName: string;
    stops: Stop[];
}

export interface Product {
    productName: string;
    productPrice: string;
    productDuration?: string;
    productValidity?: string;
    productDurationUnits?: string;
    productEndTime?: string;
}

export interface MultiOperatorGeoZoneTicket extends PeriodGeoZoneTicket {
    additionalNocs: string[];
}

export type GeoZoneTicket = PeriodGeoZoneTicket | MultiOperatorGeoZoneTicket;

export interface PeriodMultipleServicesTicket extends BasePeriodTicket {
    selectedServices: SelectedService[];
    termTime: boolean;
}

export interface MultiOperatorMultipleServicesTicket extends PeriodMultipleServicesTicket {
    additionalOperators: {
        nocCode: string;
        selectedServices: SelectedService[];
    }[];
}

export interface FlatFareTicket extends BaseTicket {
    operatorName: string;
    products: FlatFareProductDetails[];
    selectedServices: SelectedService[];
    termTime: boolean;
}

export interface BaseGroupTicket {
    nocCode: string;
    type: string;
    groupDefinition: GroupDefinition;
    email: string;
    uuid: string;
}

export interface SchemeOperatorTicket {
    schemeOperatorName: string;
    schemeOperatorRegionCode: string;
    nocCode?: string;
    type: string;
    passengerType: string;
    ageRange?: string;
    ageRangeMin?: string;
    ageRangeMax?: string;
    proof?: string;
    proofDocuments?: string[];
    email: string;
    uuid: string;
    timeRestriction: FullTimeRestriction[];
    ticketPeriod: TicketPeriod;
    products: ProductDetails[];
    zoneName: string;
    stops: Stop[];
    additionalNocs: string[];
}

export const isSchemeOperatorTicket = (data: Ticket): data is SchemeOperatorTicket =>
    (data as SchemeOperatorTicket).schemeOperatorName !== undefined &&
    (data as SchemeOperatorTicket).schemeOperatorRegionCode !== undefined;

// Matching Data (created by the user on the site)

export interface PassengerDetails {
    passengerType: string;
    ageRangeMin?: string;
    ageRangeMax?: string;
    proofDocuments?: string[];
    email: string;
    uuid: string;
    timeRestriction: FullTimeRestriction[];
    ticketPeriod: TicketPeriodWithInput;
    proof?: string[];
}

export interface CompanionInfo {
    passengerType: string;
    minNumber?: string;
    maxNumber: string;
    ageRangeMin?: string;
    ageRangeMax?: string;
    proofDocuments?: string[];
}

export interface GroupPassengerInfo extends CompanionInfo {
    ageRange: string;
    proof: string;
}

export interface GroupDefinition {
    maxGroupSize: number;
    companions: CompanionInfo[];
}

export interface TimeRestriction {
    startTime?: string;
    endTime?: string;
    validDays: ('monday' | 'tuesday' | 'wednesday' | 'thursday' | 'friday' | 'saturday' | 'sunday' | 'bankHoliday')[];
}

export interface ReturnPeriodValidity {
    amount: string;
    typeOfDuration: string;
}

export interface ReturnPeriodValidityWithErrors {
    amount?: string;
    typeOfDuration?: string;
    errors: ErrorInfo[];
}

export interface FareZone {
    name: string;
    stops: Stop[];
    prices: FareZonePrices[];
}

export interface FareZoneWithErrors {
    errors: ErrorInfo[];
}

export interface FareZonePrices {
    price: string;
    fareZones: string[];
}

export interface SelectedService {
    lineName: string;
    serviceCode: string;
    startDate: string;
    serviceDescription: string;
}

export interface BasicService {
    lineName: string;
    nocCode: string;
    operatorShortName: string;
    serviceDescription: string;
}

export interface SalesOfferPackageInfo {
    purchaseLocations: string[];
    paymentMethods: string[];
    ticketFormats: string[];
}

export interface SalesOfferPackageInfoWithErrors extends SalesOfferPackageInfo {
    errors: ErrorInfo[];
}

export interface SalesOfferPackage extends SalesOfferPackageInfo {
    name: string;
    description: string;
    id?: string;
}

export interface SalesOfferPackageWithErrors extends SalesOfferPackage {
    errors: ErrorInfo[];
}

export interface TicketPeriod {
    startDate?: string;
    endDate?: string;
}

export interface TicketPeriodWithInput extends TicketPeriod {
    dateInput: ProductDateInformation;
}

export interface BaseProduct {
    salesOfferPackages: SalesOfferPackage[];
}

export interface ProductWithSalesOfferPackages extends BaseProduct {
    productName: string;
}

export interface FlatFareProductDetails extends BaseProduct {
    productName: string;
    productPrice: string;
}

export interface ProductData {
    products: Product[];
}

export interface MultiOperatorInfoWithErrors {
    multiOperatorInfo: MultiOperatorInfo[];
    errors: ErrorInfo[];
}

export interface TimeBand {
    startTime: string;
    endTime: string;
}

export interface FullTimeRestriction {
    day: string;
    timeBands: TimeBand[];
}

export interface FullTimeRestrictionAttribute {
    fullTimeRestrictions: FullTimeRestriction[];
    errors: ErrorInfo[];
}

export interface TimeInput {
    timeInput: string;
    day: string;
}
export interface ProductDetails extends Product, BaseProduct {}
export interface FareStage {
    stageName: string;
    prices: {
        price: string;
        fareZones: string[];
    }[];
}

export interface UserFareStages {
    fareStages: FareStage[];
}

export interface UserFareZone {
    FareZoneName: string;
    NaptanCodes: string;
    AtcoCodes: string;
}

export interface DefinePassengerTypeWithErrors extends CompanionInfo {
    errors: ErrorInfo[];
}

export interface TimeRestrictionsDefinition extends TimeRestriction {
    timeRestrictionChoice?: string;
}

export interface TimeRestrictionsDefinitionWithErrors extends TimeRestrictionsDefinition {
    errors: ErrorInfo[];
}

export interface FareType {
    fareType: string;
}

export interface FareTypeWithErrors {
    errors: ErrorInfo[];
}

export interface GroupPassengerTypesCollection {
    passengerTypes: string[];
}
export interface GroupPassengerTypesCollectionWithErrors {
    errors: ErrorInfo[];
}

export interface MultiProduct {
    productName: string;
    productNameId: string;
    productPrice: string;
    productPriceId: string;
    productDuration: string;
    productDurationId: string;
    productDurationUnits: string;
    productDurationUnitsId: string;
    productValidity: string;
    productValidityId: string;
    productEndTime?: string;
    productEndTimeId?: string;
}

export interface MultiProductWithErrors extends MultiProduct {
    productNameError?: string;
    productPriceError?: string;
    productDurationError?: string;
    productDurationUnitsError?: string;
}

export interface PassengerType {
    passengerType: string;
    ageRange?: string;
    ageRangeMin?: string;
    ageRangeMax?: string;
    proof?: string;
    proofDocuments?: string[];
}

export interface PassengerTypeWithErrors {
    errors: ErrorInfo[];
}

export interface FaresInput {
    input: string;
    locator: string;
}

export interface FaresInformation {
    inputs: FaresInput[];
    errorInformation: PriceEntryError[];
}

export interface TicketPeriodWithErrors {
    errors: ErrorInfo[];
    dates: ProductDateInformation;
}

export interface ProductDateInformation {
    startDateDay: string;
    startDateMonth: string;
    startDateYear: string;
    endDateDay: string;
    endDateMonth: string;
    endDateYear: string;
}

export interface SelectSalesOfferPackageWithError {
    errors: ErrorInfo[];
    selected: { [key: string]: string[] };
}

export interface Service {
    service: string;
}

export interface ServiceWithErrors {
    errors: ErrorInfo[];
}

export interface ServiceType {
    lineName: string;
    startDate: string;
    description: string;
    origin?: string;
    destination?: string;
    serviceCode: string;
    dataSource?: string;
}

export interface ServicesInfo extends ServiceType {
    checked?: boolean;
}

export interface ServiceDB {
    serviceDescription: string;
    operatorShortName: string;
    journeyPatterns: JourneyPattern[];
}

export interface RawService {
    serviceDescription: string;
    operatorShortName: string;
    journeyPatterns: RawJourneyPattern[];
}

export interface JourneyPattern {
    startPoint: {
        Id: string;
        Display: string;
    };
    endPoint: {
        Id: string;
        Display: string;
    };
    stopList: string[];
}

export interface RawJourneyPattern {
    orderedStopPoints: {
        stopPointRef: string;
        commonName: string;
    }[];
}

// Components

export interface ConfirmationElement {
    name: string;
    content: string;
    href: string;
}

export interface FareTypeRadio {
    fareType: string;
    label: string;
}
export interface FareTypeRadioProps {
    standardFares: FareTypeRadio[];
    otherFares?: FareTypeRadio[];
}

export interface RadioWithoutConditionals extends BaseReactElement {
    value: string;
    radioButtonHint?: {
        id: string;
        content: string;
    };
}

export interface RadioWithConditionalInputs extends RadioWithoutConditionals {
    dataAriaControls: string;
    inputHint: {
        id: string;
        content: string;
        hidden?: boolean;
    };
    inputType: 'text' | 'checkbox' | 'date' | 'textWithUnits' | 'dropdown';
    inputs: BaseReactElement[] | PremadeTimeRestriction[];
    inputErrors: ErrorInfo[];
    selectIdentifier?: string;
}

export type RadioButton = RadioWithoutConditionals | RadioWithConditionalInputs;

export interface RadioConditionalInputFieldset {
    heading: {
        id: string;
        content: string;
        hidden?: boolean;
    };
    radios: RadioButton[];
    radioError: ErrorInfo[];
}

export interface UserAttribute {
    redirectFrom?: string;
}

export interface OperatorAttribute {
    name?: string;
    region?: string;
    nocCode?: string;
    uuid?: string;
    email?: string;
}

export interface ForgotPasswordAttribute {
    email: string;
}

export interface TxcSourceAttribute {
    source: 'tnds' | 'bods';
    hasTnds: boolean;
    hasBods: boolean;
}
