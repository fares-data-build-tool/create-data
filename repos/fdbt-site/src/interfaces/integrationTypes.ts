export type ExportLambdaBody = { paths: string[]; noc: string; exportPrefix?: string };

export type ZipperLambdaBody = { exportName: string; noc: string };
