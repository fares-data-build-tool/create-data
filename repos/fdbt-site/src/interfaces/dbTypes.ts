import { CompanionInfo, DbTimeBand, TimeRestrictionDay } from './matchingJsonTypes';

export interface FullGroupPassengerType {
    id: number;
    name: string;
    groupPassengerType: GroupPassengerType;
}

export interface SinglePassengerType {
    id: number;
    name: string;
    passengerType: PassengerType;
}

export interface PassengerType {
    passengerType: string;
    ageRange?: string;
    ageRangeMin?: string;
    ageRangeMax?: string;
    proof?: string;
    proofDocuments?: string[];
    id: number;
}

export interface MyFaresProduct {
    id: number;
    lineId: string;
    matchingJsonLink: string;
    startDate: string;
    endDate?: string;
    servicesRequiringAttention?: string[];
    fareTriangleModified?: string;
}

export type RawMyFaresProduct = Omit<MyFaresProduct, 'servicesRequiringAttention'> & {
    servicesRequiringAttention: string;
};

export interface DbProduct {
    id: string;
    matchingJsonLink: string;
    lineId: string;
    startDate: string;
    endDate?: string;
}

export type MyFaresOtherProduct = Omit<MyFaresProduct, 'lineId'>;

export interface ServiceQueryData {
    operatorShortName: string;
    serviceDescription: string;
    lineName: string;
    startDate: string;
    lineId: string;
    fromAtcoCode: string;
    toAtcoCode: string;
    fromCommonName: string;
    toCommonName: string;
    journeyPatternId: string;
    order: string;
    direction: string;
    fromSequenceNumber: number;
    toSequenceNumber: number;
    inboundDirectionDescription: string;
    outboundDirectionDescription: string;
}

export type StopPoint = {
    stopPointRef: string;
    commonName: string;
    sequenceNumber?: number;
};

export interface RawJourneyPattern {
    orderedStopPoints: StopPoint[];
    direction: string;
}

export interface RawService {
    serviceDescription: string;
    operatorShortName: string;
    journeyPatterns: RawJourneyPattern[];
    lineId: string;
    lineName: string;
    startDate: string;
    endDate?: string;
    inboundDirectionDescription: string;
    outboundDirectionDescription: string;
}

export interface GroupPassengerType {
    name: string;
    maxGroupSize: string;
    companions: CompanionInfo[];
}

export interface RawSalesOfferPackage {
    name: string;
    description: string;
    id: number;
    purchaseLocations: string;
    paymentMethods: string;
    ticketFormats: string;
}
export interface GroupPassengerTypeDb {
    id: number;
    name: string;
    groupPassengerType: GroupPassengerTypeReference;
}

export interface CompanionReference {
    id: number;
    name: string;
    minNumber?: string;
    maxNumber: string;
}

export interface GroupPassengerTypeReference {
    name: string;
    maxGroupSize: string;
    companions: CompanionReference[];
}

export interface DbTimeRestriction {
    day: TimeRestrictionDay;
    timeBands: DbTimeBand[];
}

export interface ServiceDetails {
    fromAtcoCode: string;
    toAtcoCode: string;
    direction: string;
    serviceId: string;
}
