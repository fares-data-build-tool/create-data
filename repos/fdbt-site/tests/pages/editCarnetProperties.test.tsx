import { shallow } from 'enzyme';
import React from 'react';
import { CarnetExpiryUnit } from '../../src/interfaces/matchingJsonTypes';
import EditCarnetProperties from '../../src/pages/editCarnetProperties';

describe('pages', () => {
    describe('editCarnetProperties', () => {
        it('should render editCarnetProperties page correctly', () => {
            const wrapper = shallow(
                <EditCarnetProperties
                    errors={[]}
                    csrfToken=""
                    expiryTime="3"
                    expiryUnit={CarnetExpiryUnit.HOUR}
                    quantity="3"
                />,
            );
            expect(wrapper).toMatchSnapshot();
        });
    });
});
