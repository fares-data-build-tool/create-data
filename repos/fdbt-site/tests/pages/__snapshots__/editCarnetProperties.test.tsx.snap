// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages editCarnetProperties should render editCarnetProperties page correctly 1`] = `
<FullColumnLayout
  description="Edit Carnet Properties page of the Create Fares Data Service"
  errors={Array []}
  title="Edit Carnet Properties - Create Fares Data Service"
>
  <CsrfForm
    action="/api/editCarnetProperties"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={Array []}
    />
    <h1
      className="govuk-heading-l"
      id="edit-period-duration-page-heading"
    >
      Edit Carnet Properties
    </h1>
    <div
      className="flex-container"
    >
      <div
        className="govuk-!-margin-right-2"
      >
        <label
          className="govuk-label"
          htmlFor="edit-carnet-quantity"
        >
          <span
            aria-hidden={true}
          >
            Quantity in bundle
          </span>
        </label>
        <span
          className="govuk-hint"
          id="edit-carnet-quantity-hint"
        >
          Must be 2 or more
        </span>
        <input
          aria-describedby="edit-carnet-quantity-hint"
          className="govuk-input govuk-input--width-6"
          data-non-numeric={true}
          defaultValue="3"
          id="edit-carnet-quantity"
          name="carnetQuantityInput"
          type="text"
        />
      </div>
      <div
        className="govuk-!-margin-left-2 govuk-!-margin-right-2"
      >
        <label
          className="govuk-label"
          htmlFor="edit-carnet-expiry"
        >
          Carnet expiry
        </label>
        <span
          className="govuk-hint"
          id="edit-carnet-expiry-hint"
        >
          e.g. 2 months
        </span>
        <ExpirySelector
          carnet={true}
          defaultDuration="3"
          defaultUnit="hour"
          errors={Array []}
          hideFormGroupError={true}
          hintId="edit-carnet-expiry-hint"
          quantityId="edit-carnet-expiry-duration"
          quantityName="carnetExpiryDurationInput"
          unitId="edit-carnet-expiry-unit"
          unitName="carnetExpiryUnitInput"
        />
      </div>
    </div>
    <br />
    <input
      className="govuk-button"
      id="continue-button"
      type="submit"
      value="Continue"
    />
  </CsrfForm>
</FullColumnLayout>
`;
